Validation Checks

lognath-ebi opened the pull request:
The PR has the following basic checks implemented on Validation
- LengthValidation on features
- DuplicateValidation on feature proteinId
- LocationValidation on features (start and end)

High level implementation with basic validations and unit test cases, if this looks fine I will follow the similar approach for the rest.

Rajkumar-D commented on src/main/java/uk/ac/ebi/embl/converter/validation/builtin/LengthValidation.java:
```diff
@@ -0,0 +1,38 @@
+        String featureName = feature.getName();
+        long length = feature.getLength();
+
+        if (featureName.equalsIgnoreCase(Feature.PROPETIDE_FEATURE_NAME) && feature.getLength() % 3 != 0) {
```
Since ontology terms change only rarely, we can define them as constants in Gff3Anthology. If the SO term changes in the CSV, we must also update the corresponding constant. To ensure consistency, we can add a Validation Gff3AnthologyValidator that cross-checks the constants against the ontology values in the CSV.

lognath-ebi commented on src/main/java/uk/ac/ebi/embl/converter/validation/builtin/LocationValidation.java:
```diff
@@ -0,0 +1,83 @@
+                    "The start & end values for accession \"%s\" is invalid".formatted(feature.accession()));
+        }
+
+        if (Feature.PROPETIDE_FEATURE_NAME.equals(feature.getName())) {
```
This isn’t a finished one, I was working on it and accidentally pushed the inprogress code. I’ll be handling this at the annotation level.

lognath-ebi replied:
This isn’t a finished version, I am still working on it and accidentally pushed the inprogress code. I’ll be handling this at the annotation level

lognath-ebi commented:
Addressed the PR review comments - Regarding the propeptide validations, need to get the clarification from Peter. Based on that, we can go for Feature level or Annotation level validations approach. Currently, I kept the validations in Annotation level.

lognath-ebi commented on src/main/java/uk/ac/ebi/embl/converter/validation/builtin/DuplicateFeatureValidation.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Override
+    public void validateAnnotation(GFF3Annotation gff3Annotation, int line) throws ValidationException {
```
yes, I have moved this validation to feature level.
